### PR TITLE
remove dup code for section.reader access

### PR DESCRIPTION
**Pull request by half-life666**
# What problem are we solving?
https://github.com/seaweedfs/seaweedfs/issues/6168

# How are we solving the problem?
resolve the thread safty for function

# How is the PR tested?
not yet, will do long IO test

# Checks
- [ ] I have added unit tests if possible.
- [ ] I will add related wiki document changes and link to this PR after merging.

### CODE REVIEW

**Comment by half-life666** [comment]
@chrislusf can you have a look at this?

**Comment by chrislusf** [comment]
* how is the long IO test?
* did it affect performance? https://github.com/seaweedfs/seaweedfs/pull/4400 intended to add this for performance.

**Comment by half-life666** [comment]
> * how is the long IO test?
> 
>     * did it affect performance? [Mount concurrent read #4400](https://github.com/seaweedfs/seaweedfs/pull/4400) intended to add this for performance.

Still running, for ~3 days. For performance, I need to take a look at 4400 first.

**Comment by half-life666** [comment]
As in https://github.com/seaweedfs/seaweedfs/issues/6168, I thought the dup code is just a mistake, I don't see any reason to performace same operation before and after lock acquire:


```go

func (section *FileChunkSection) setupForRead(group *ChunkGroup, fileSize int64) {
	if section.isPrepared {
		section.reader.fileSize = fileSize
		return
	}

	section.lock.Lock()
	defer section.lock.Unlock()

	if section.isPrepared {
		section.reader.fileSize = fileSize
		return
	}
```

File: weed/shell/command_ec_encode.go
Comment (chrislusf) [inline weed/shell/command_ec_encode.go:1]: update from master and this change is not needed any more.

Comment (half-life666) [reply weed/shell/command_ec_encode.go:1]: removed

**Comment by half-life666** [comment]
Long IO is running for >7 days now

**Pull request merged**